src/pacasam/extractors/bd_ortho_vintage.py

```diff
@@ -44,64 +47,69 @@ class BDOrthoVintageExtractor(Extractor):
     """
 
     patch_suffix: str = ".tiff"
-    rgb_column: str = "rgb_file"
-    irc_column: str = "irc_file"
-    pixel_per_meter: int = 5
 
     def extract(self) -> None:
         """Download the orthoimages dataset."""
         iterable_of_args = []
-        for (rgb_file, irc_file), single_imagery in self.sampling.groupby([self.rgb_column, self.irc_column]):
+        for (rgb_file, irc_file), single_imagery in self.sampling.groupby([RGB_COLNAME, IRC_COLNAME]):
             iterable_of_args.append((rgb_file, irc_file, single_imagery))
 
         with WorkerPool(n_jobs=self.num_jobs) as pool:
             pool.map(self.extract_from_single_file, iterable_of_args, progress_bar=True)
 
     def extract_from_single_file(self, rgb_file, irc_file, single_file_sampling: GeoDataFrame):
-        with rasterio.open(rgb_file) as rgb, rasterio.open(irc_file) as irc:
+        with rasterio.open(rgb_file) as rgb_open, rasterio.open(irc_file) as irc_open:
```

Comment:
I'm not sure I understand why you pass the file handle to `extract_rngbir_patch_as_tmp_file` and not just the path (then using the context manager inside this function), does it save time or make it somewhat safer to open it only once?

Reply:
I thought it would save time but it probably does not. I had the misconception that opening the files would read them entirely, which is probably not the case.

Reply:
Updated to reopen everytime, simplifying code structure. And changed parallelization unit from file to patch since we reopen files anyway :)